Gems/Atom/Utils/Code/Include/Atom/Utils/ImGuiCpuProfiler.h

```diff
@@ -159,10 +151,34 @@ namespace AZ
             // Tracks the frame boundaries
             AZStd::vector<AZStd::sys_time_t> m_frameEndTicks = { INT64_MIN };
 
-            // Main data structure for storing function statistics to be shown in the popup windows.
-            // For now we default allocate for all regions on the first render frame and then use RegionStatistics.m_draw to determine
-            // if we should draw the window or not. FIXME(ATOM-15948) this should be changed once RegionStatistics gets heavier. 
-            AZStd::unordered_map<const GroupRegionName*, RegionStatistics> m_regionStatisticsMap;
+            // Filter for highlighting regions on the visualizer
+            ImGuiTextFilter m_visualizerHighlightFilter;
+
+            // --- Tabular view members --- 
+
+            // ImGui filter used to filter TimedRegions.
+            ImGuiTextFilter m_timedRegionFilter;
+
+            // Saves statistical view data organized by group name -> region name -> row data
+            GroupRegionMap m_groupRegionMap;
+
+            // Saves pointers to objects in m_groupRegionMap, order reflects table ordering. 
+            // Non-owning, will be cleared when m_groupRegionMap is cleared. 
+            AZStd::vector<TableRow*> m_tableData;
```

Comment:
Question: is there an issue with using raw pointers here? They point to objects in `m_groupRegionMap`, so we don't have to worry about deallocation. We also clear `m_tableData` whenever `m_groupRegionMap` is cleared. The only issue is if inserts into `m_groupRegionMap` invalidate pointers into it, which I don't think is the case - I can also try moving to an `AZStd::vector<reference_wrapper ...>` as a safer alternative.

Reply:
Thought about this a little more over the weekend, I think as long as we make sure that elements don't get deleted then this should be safe. Since `AZStd::map` is backed by an rbtree, the locations of objects should stay the same - inserts will only change the left/right pointers of a node which we don't save here.